lib/apollo-federation/schema.rb

```diff
@@ -12,9 +12,27 @@ def self.included(klass)
     end
 
     module CommonMethods
+      FEDERATION_2_PREFIX = <<~SCHEMA
+        extend schema
+          @link(url: "https://specs.apollo.dev/federation/v2.0",
+          import: ["@key", "@extends", "@external", "@requires", "@provides", "@shareable", "@inaccessible", "@override"])
```

Comment:
This looks like it's importing *all* of the directives unconditionally. Is that acceptable, or should it be more dynamic to only import the directives that are actually used in the schema?

[Apollo docs](https://www.apollographql.com/docs/federation/federated-types/federated-directives/#importing-directives) say "Modify the import array to include whichever federated directives your subgraph schema uses."

Reply:
Is `@extends` a valid directive in this context? I don't see it in the docs.

Reply:
One subtle feature of the `@link` spec is that you don't _have_ to import these directives. If you don't supply the `import:` argument, the directives are still available in a namespaced form:

```graphql 
extend schema @link(url: "https://specs.apollo.dev/federation/v2.0")

type Foo @federation__key(fields: "bar") @federation__shareable {
  bar: ID!
  baz: String @federation__external
  quux: String @federation__requires(fields: "bax")
  blah: String @federation__inaccesible
}
```

This might be the "right" approach for a code-first library. The `key` ruby method could output `@federation__key`.

Reply:
And yes, I believe the `@extends` directive was dropped entirely in Fed 2.

Reply:
I think `@extends` is still optionally supported. We either need to support `@extends` or have proper support for type extensions (eg.  `extend type`).
See: https://www.apollographql.com/docs/federation/federation-spec/#federation-schema-specification

Reply:
> This looks like it's importing _all_ of the directives unconditionally. Is that acceptable, or should it be more dynamic to only import the directives that are actually used in the schema?
> 
> [Apollo docs](https://www.apollographql.com/docs/federation/federated-types/federated-directives/#importing-directives) say "Modify the import array to include whichever federated directives your subgraph schema uses."

I did consider trying to make this dynamic but it was complicated to implement. In the end, I couldn't think of a use case where this would actually cause a problem so I figured it was ok to include all the directives. @lyricsboy Do you have an example of how this could be a problem?

@lennyburdette - Not a big fan of using the `@federation__external` approach as I've never seen this mentioned in any of the docs. I fear it would cause some confusion when the schema we're generating doesn't match what people see in the docs. If we were to take this approach, should it apply to v1 and v2 or only v2?

Reply:
The way this is done currently, @lennyburdette , does it pass the `@link` test in https://www.apollographql.com/docs/federation/other-servers/?

Reply:
Yeah this should be sufficient. The test is rudimentary right now: it just searches the `_Service.sdl` for the presence of "@link": https://github.com/apollographql/apollo-federation-subgraph-compatibility/blob/main/src/tests/link.test.ts